07_errors/patrickmarabeas/mapStore.go

```diff
@@ -9,43 +9,50 @@ func NewMapStore() Storer {
 }
 
 // Create increments the uuid and adds the provided Puppy struct to the store with this identifier.
-func (store *MapStore) Create(puppy Puppy) int {
+func (store *MapStore) Create(puppy Puppy) (int, error) {
+	if puppy.Value < 0 {
+		return -1, NewError(NegativeValue)
+	}
+
 	puppy.ID = store.uuid
 	store.store[puppy.ID] = puppy
 	store.uuid++
 
-	return puppy.ID
+	return puppy.ID, nil
 }
 
 // Read returns the puppy matching the provided uuid.
 // An empty Puppy struct is returned if the identifier does not exist.
-func (store *MapStore) Read(id int) Puppy {
+func (store *MapStore) Read(id int) (Puppy, error) {
 	if _, ok := store.store[id]; ok {
-		return store.store[id]
+		return store.store[id], nil
 	}
 
-	return Puppy{}
+	return Puppy{}, NewError(IDNotFound)
```

Comment:
Two comments about this function (and I know my comments really apply to lab 6, not this one but here we are now):
1) Usually you want to align the happy path to the left edge, so the indented case should be returning the error,
2) Since you have the non-error path inside the if, rather than indexing the map again you should save the value rather than using the blank identifier:
```
if p, ok := store.store[id]; ok {
    return p, nil
}
```

unless you use a `var` to declare `p` outside the scope of the `if`. Combined, it would look like:
```
func (store *MapStore) Read(id int) (Puppy, error) {
    var p Puppy
    if p, ok := store.store[id]; !ok {
        return Puppy{}, NewError(IDNotFound)
    }
    return p, nil
}
```